src/meepgeom.cpp

```diff
@@ -307,6 +307,70 @@ bool is_metal(meep::field_type ft, const material_type *material) {
     }
 }
 
+/* implement mirror boundary conditions for i outside 0..n-1: */
+static int mirrorindex(int i, int n) { return i >= n ? 2 * n - 1 - i : (i < 0 ? -1 - i : i); }
```

Comment:
Looks like this is already defined in

https://github.com/NanoComp/meep/blob/899f6d039346dd9837d1142f48c5cceed595cb56/src/meepgeom.cpp#L2377

Reply:
Ah I see you moved it. Maybe locate all the gradient codes next to each other?

Reply:
Fixed.